sklearn/cluster/_hdbscan/_tree.pyx

```diff
@@ -282,40 +282,38 @@ cdef max_lambdas(cnp.ndarray hierarchy):
     return deaths
 
 
-cdef class TreeUnionFind (object):
+cdef class TreeUnionFind:
```

Comment:
If this not extended and is not meant to, we can decorate it this `final`:

```suggestion
@final
cdef class TreeUnionFind:
```

What do you think?